system/include/emscripten/val.h

```diff
@@ -385,6 +401,17 @@ namespace emscripten {
             : handle(internal::_emval_new_cstring(v))
         {}
 
+#ifndef _LIBCPP_HAS_NO_CHAR8_T
```

Comment:
This seems rather odd.. can we not do a positive check?  Maybe just skip this and stick to `const char*` for now ..  unless there is some actual benefit?  (I don't think there is, is there?)

Reply:
Oh, a bit of weird. I just searched the repo and got it from here:
https://github.com/emscripten-core/emscripten/blob/main/system/lib/libcxx/include/__config#L987
> #if _LIBCPP_STD_VER <= 17 || !defined(__cpp_char8_t)
> #define _LIBCPP_HAS_NO_CHAR8_T
> #endif
We can use the standard `__cpp_char8_t` if necessary.  But I agree with you that `const char*` is good for now.  We can do this later when need.  I'll revert the second commit.